machine/v1beta1/types_vsphereprovider.go

```diff
@@ -87,12 +87,61 @@ type NetworkSpec struct {
 	Devices []NetworkDeviceSpec `json:"devices"`
 }
 
+// AddressesFromPool is an IPAddressPool that will be used to create
+// IPAddressClaims for fulfillment by an external controller.
+type AddressesFromPool struct {
+	// group of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Group string `json:"group"`
+	// resource of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Resource string `json:"resource"`
+	// name of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
```

Comment:
Same as above, we can add a regex for this, might be worth looking for other examples of name validations and group/resource validations within openshift/api

Reply:
I made changes based on a couple of other examples in openshift/api